rfcs/text/0000-planning-for-the-dcp.md

```diff
@@ -0,0 +1,221 @@
+
+New items in the *dcp* repo appear in the ZenHub *New* pipeline and MUST be triaged into the *Product Backlog* pipeline, the *Icebox* pipeline, or the *Closed* pipeline.
+
+The ZenHub *Product Backlog* pipeline is the top-level DCP product backlog which is ordered during Refinement.
+
+The ZenHub *Icebox* pipeline SHOULD be used with great constraint in the *dcp* repo. It's not a pipeline for avoiding hard decisions during Refinement. It's intended to document potential *futures* from the DCP roadmap. Individual component repo(s) MAY use the *Icebox* pipeline in a manner of their own choosing.
+
+No issues in the *dcp* repo MUST be assigned to the *Epic* pipeline. Individual component repo(s) MAY use the *Epic* pipeline in a manner of their own choosing.
+
+### Sprint Planning
+
+Prior to the start of the DCP Sprint, component teams MUST identify their *Sprint Backlog* - a slice of items from the the ZenHub *Product Backlog* pipeline in their repo which are assigned to the ZenHub Milestone for the Sprint and moved to the ZenHub *Sprint Backlog* pipeline.
+
+The number of items selected is solely up to the Development Team. 
+- [ZenHub story points](https://help.zenhub.com/support/solutions/articles/43000034282) MUST be assigned to all *Sprint Backlog* items. (Story points are required to enable [ZenHub burndown charts](https://help.zenhub.com/support/solutions/articles/43000010356-track-sprint-progress-with-burndown-charts) to track the progress of a Sprint.)
+  - [How to start?](https://www.scrum.org/forum/scrum-forum/6774/how-decide-how-many-story-points-take-first-sprint)
+- Research and prototyping tasks SHOULD be modeled as Spikes.
+- Dependencies between components within a Sprint MUST be communicated between the respective teams and surfaced using [ZenHub Dependencies](https://help.zenhub.com/support/solutions/articles/43000010349). 
+
+At the start of the Sprint, the single DCP Sprint Backlog SHOULD be reviewed by the **PM Team** and **Architecture Team** in their oversight roles.
+
+### Sprint(ing)
+
+During Development, the Development Team assigns issues from the ZenHub *Sprint Backlog* pipeline and move to the *In Progress* pipeline. 
+
+When a pull request is filed and in review for an *In Progress* item, the issue is moved to the *Review* pipeline. 
+
+If the pull request is rejected, the issue returns to *In Progress*.
+
+Otherwise, the pipeline tracks the issue through the DCP [promotion sequence](https://allspark.dev.data.humancellatlas.org/dcp-ops/docs/wikis/SOP:%20Releasing%20new%20Versions%20of%20DCP%20Software).
+
+The approved pull request is merged to the default branch, deployed to Development, and the issue moves to the *Development* pipeline.
+
+When the changes are promoted to Integration, the issue moves to the *Integration* pipeline.
+
+When the changes are promoted to Staging, the issue moves to the *Staging* pipeline.
+
+When the changes are promoted to Production, the issue moves to the *Production* pipeline.
+
```

Comment:
This creates a lot of (manual) overhead to move whole batches of issues from Dev to Integration to Staging to Production as each promotion occurs. Also a sprint can't be closed until all issues in the sprint are closed, which could be weeks (or months) later. Perhaps issues could be verified/closed in Dev (as now) and what's deployed in Integration, Staging and Production could be tracked via the sofware version numbers deployed there - someone looking at a particular issue could see what software release it is in and compare with the software releases in the various environments to see where it has bene deployed so far.

Reply:
Regarding sprint closing, I think this could be one of those examples where we are trying to bend to a process rather than what's useful on a development level.